docs/gettinginvolved/infrastructure/documentation.rst

```diff
@@ -0,0 +1,182 @@
+Documentation Hosting
+=====================
+
+Flocker's documentation is hosted on S3, fronted by CloudFront.
+
+S3 Buckets
+----------
+
+There are 3 S3 buckets used for documentation.
+
+``clusterhq-docs``
+~~~~~~~~~~~~~~~~~~
+
+This bucket hosts our public facing documentation.
+
+It has documentation for all marketing releases.
+
+Configuration
+`````````````
+It is configured to allow static website hosting, with an index document of ``index.html`` and an error document of ``404.html``.
+To allow deep-linking to the latest documentation, the following redirect configuration is
+specified (replace the versions with the latest marketing and development releases).
+The latest marketing and development releases are redirected with temporary redirects to the most recent version;
+obsolete releases that are no longer hosted are permanently redirected to the latest version.
+
+.. code:: xml
+
+   <RoutingRules>
+     <!-- Marketing Release Documentation -->
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/latest/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/0.3.2/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>302</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+     <!-- Development Release Documentation -->
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/devel/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/0.3.3dev3/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>302</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+     <!-- Obsolete Release Documentation -->
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/0.3.1/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/latest/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>301</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/0.3.0/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/latest/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>301</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/0.2.1/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/latest/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>301</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+     <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>en/0.1.2/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>en/latest/</ReplaceKeyPrefixWith>
+         <HttpRedirectCode>301</HttpRedirectCode>
+       </Redirect>
+     </RoutingRule>
+   </RoutingRules>
+
+To allow CloudFront to access the bucket, it has the following bucket policy configured:
```

Comment:
is this an example of the correct configuration or is it the canonical source?  It looks pretty comprehensive - but the release process documentation only suggests updating the version stored on AWS, not this version.  So I guess it's just an example - maybe be more explicit about that.

Later, I suggest we go with something like a simpler file describing releases - like:

```
{"latest: "0.3.2", "latest-dev": "0.3.3dev3", "obsolete": ["0.3.0", "0.2.1", "0.1.2"]}
```

that is kept up to date as part of the release process and a tool that generates the routing rules from this data and updates AWS with the new version.  If that seems sound to you, file a ticket and link to it from here-ish?